pandas/core/series.py

```diff
@@ -3626,9 +3626,14 @@ def _gotitem(self, key, ndim, subset=None):
         **_shared_doc_kwargs,
     )
     @Appender(generic._shared_docs["aggregate"])
-    def aggregate(self, func, axis=0, *args, **kwargs):
+    def aggregate(self, func=None, axis=0, *args, **kwargs):
         # Validate the axis parameter
         self._get_axis_number(axis)
+
+        if func is None:
+            func = dict(kwargs.items())
```

Comment:
why are you doing this?

Reply:
this is needed, for cases like this `df.B.agg(foo=sum, bar="min", cat="min")`, this has to be converted so that `func` can be understood.

i removed the line below

Reply:
can you add a comment here giving this explanation

Reply:
can you update this

Reply:
yeah, indeed, it's not ideal, it was inspired by https://github.com/pandas-dev/pandas/pull/26399 , and set `func=None` by default in order to use named aggregation where `agg(foo=('a', min), bar=('b', max))` can be parsed later.

since we haven't seen many bug reports about this renaming enhancement in `df.groupby.agg`, so maybe we could let this roll out for `df.agg` and see how it goes?